src/python/WMCore/WMSpec/WMWorkloadTools.py

```diff
@@ -44,6 +44,20 @@ def makeList(stringList):
     raise WMSpecFactoryException("Can't convert to list %s" % stringList)
 
 
+def makeNonEmptyList(stringList):
```

Comment:
I'd rather see this in Utils (perhaps true of everything here) with a generic exception. But I think there is already exactly this in WMObject if you are inheriting from that. 

Having this general purpose stuff scattered around is a PITA.